lib/jsdom/browser/Window.js

```diff
@@ -168,70 +171,40 @@ function Window(options) {
     return result;
   };
 
-  this.XMLHttpRequest = function () {
-    var xhr = new XMLHttpRequest();
-    var lastUrl = "";
-    xhr._open = xhr.open;
-    xhr.open = function (method, url, async, user, password) {
-      lastUrl = fixUrlForBuggyXhr(resolveHref(window.document.URL, url));
-      return xhr._open(method, lastUrl, async, user, password);
-    };
-    xhr._getAllResponseHeaders = xhr.getAllResponseHeaders;
-    xhr.getAllResponseHeaders = function () {
-      if (lastUrl.startsWith("file:")) {
-        // Monkey patch this function for files. The node-xhr module will crash for file URLs.
-        return null;
-      }
-      return xhr._getAllResponseHeaders();
-    };
-    xhr._send = xhr.send;
-    xhr.send = function (data) {
-      var cookieJar = window.document._cookieJar;
-      var cookieStr = cookieJar.getCookieStringSync(lastUrl, {http: true});
-
-      if (cookieStr) {
-        xhr.setDisableHeaderCheck(true);
-        xhr.setRequestHeader("cookie", cookieStr);
-        xhr.setDisableHeaderCheck(false);
-      }
+  this.File = file.File;
+  this.Blob = file.Blob;
 
-      function setReceivedCookies() {
-        if (xhr.readyState === xhr.HEADERS_RECEIVED) {
-          var receivedCookies = xhr.getResponseHeader("set-cookie");
+  this.FileReader = file.createFileReader({
+    window: window
+  });
 
-          if (receivedCookies) {
-            receivedCookies = Array.isArray(receivedCookies) ? receivedCookies : [receivedCookies];
+  const xhrObjs = xhr.createXMLHttpRequest({
+    window: window
+  });
 
-            receivedCookies.forEach(function (cookieStr) {
-              cookieJar.setCookieSync(cookieStr, lastUrl, {
-                http: true,
-                ignoreError: true
-              });
-            });
-          }
+  this.FormData = xhr.FormData;
 
-          xhr.removeEventListener("readystatechange", setReceivedCookies);
-        }
-      }
+  this.XMLHttpRequest = xhrObjs.XMLHttpRequest;
+  this.XMLHttpRequestUpload = xhrObjs.XMLHttpRequestUpload;
 
-      xhr.addEventListener("readystatechange", setReceivedCookies);
-      return xhr._send(data);
-    };
-    Object.defineProperty(xhr, "response", {
-      get: function () {
-        if (this.responseType === "text" || !this.responseType) {
-          // Spec says "text" or "", but responseType support is incomplete, so we need to catch more cases.
-          return this.responseText;
-        } else if (this.responseType === "json") {
-          return JSON.parse(this.responseText);
-        } else {
-          return null; // emulate failed request
-        }
-      },
-      enumerable: true,
-      configurable: true
-    });
-    return xhr;
+  this.ArrayBuffer = ArrayBuffer;
```

Comment:
Note that I have not added some tests because of "instanceof Function" which does not work with classes/functions created outside the Window context. (Blob instanceof Function, Window instanceof Function, etc)